src/types/log.rs

```diff
@@ -106,27 +120,33 @@ impl FilterBuilder {
     /// Sets `from_block`. The fields `from_block` and `block_hash` are
     /// mutually exclusive. Setting `from_block` will clear a previously set
     /// `block_hash`.
-    pub fn from_block(mut self, block: BlockNumber) -> Self {
-        self.filter.block_hash = None;
-        self.filter.from_block = Some(block);
+    pub fn from_block(mut self, block: Option<BlockNumber>) -> Self {
+        if let Some(block) = block {
+            self.filter.block_hash = None;
+            self.filter.from_block = Some(block);
+        }
```

Comment:
This method should not be changed, the builder should only take values that are defined, not `Option`s. `None` case should be handled at the call site.